src/MainWindow.vala

```diff
@@ -62,6 +62,17 @@ public class Sideload.MainWindow : Gtk.ApplicationWindow {
         file.progress_changed.connect (on_progress_changed);
         file.installation_failed.connect (on_install_failed);
         file.installation_succeeded.connect (on_install_succeeded);
+        file.details_ready.connect (() => {
+            if (file.already_installed) {
+                var success_view = new SuccessView (SuccessView.SuccessType.ALREADY_INSTALLED);
+
+                stack.add (success_view);
+                stack.visible_child = success_view;
+            } else {
+                main_view.display_details (file.download_size, file.already_installed);
```

Comment:
Don't need to pass `already_installed` to this method anymore :slightly_smiling_face: